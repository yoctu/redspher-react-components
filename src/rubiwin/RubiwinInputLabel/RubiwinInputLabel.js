import React from 'react'
import PropTypes from 'prop-types'
import InputLabel from '@mui/material/InputLabel'

const RubiwinInputLabel = ({
  children,
  disabled,
  error,
  htmlFor,
  className
}) => (
  <InputLabel
    disabled={disabled}
    error={error}
    htmlFor={htmlFor}
    className={className}
  >
    {children}
  </InputLabel>
)

RubiwinInputLabel.propTypes = {
  /** label text */
  children: PropTypes.node,
  /** add a className */
  className: PropTypes.string,
  /** is disabled */
  disabled: PropTypes.bool,
  /** has error state */
  error: PropTypes.bool,
  /** htmlFor */
  htmlFor: PropTypes.string
}

export default RubiwinInputLabel
