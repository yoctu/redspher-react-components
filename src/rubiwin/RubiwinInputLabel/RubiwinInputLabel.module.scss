@import '../rubiwin-constant';

.root {
  font-weight: bold;
  font-size: 14px;
  margin-left: 6px;
  margin-bottom: 6px;
}
