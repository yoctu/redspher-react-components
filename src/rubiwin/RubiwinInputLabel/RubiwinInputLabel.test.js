import React from 'react'
import renderer from 'react-test-renderer'
import RubiwinInputLabel from './RubiwinInputLabel'

const testProps = {
  children: 'test text',
  htmlFor: 'textId',
  className: 'addClass'
}

/** Snapshots */
it('renders correctly props object', () => {
  const tree = renderer.create(<RubiwinInputLabel {...testProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('has error', () => {
  const tree = renderer.create(<RubiwinInputLabel error />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('is disabled', () => {
  const tree = renderer.create(<RubiwinInputLabel disabled />).toJSON()
  expect(tree).toMatchSnapshot()
})
