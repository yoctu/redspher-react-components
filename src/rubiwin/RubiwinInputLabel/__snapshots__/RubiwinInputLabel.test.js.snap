// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has error 1`] = `
<label
  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
/>
`;

exports[`is disabled 1`] = `
<label
  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
/>
`;

exports[`renders correctly props object 1`] = `
<label
  className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary addClass css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
  htmlFor="textId"
>
  test text
</label>
`;
